plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -0,0 +1,199 @@
+// It looks at all new pods and sets pod.Spec.Overhead if a RuntimeClass is specified which
+// defines an Overhead. TODO: add blurb about validation/rejection handling
+type RuntimeClass struct {
+	*admission.Handler
+	client kubernetes.Interface
+	lister nodev1beta1listers.RuntimeClassLister
+}
+
+var _ admission.MutationInterface = &RuntimeClass{}
+var _ admission.ValidationInterface = &RuntimeClass{}
+
+var _ genericadmissioninitailizer.WantsExternalKubeInformerFactory = &RuntimeClass{}
+var _ genericadmissioninitailizer.WantsExternalKubeClientSet = &RuntimeClass{}
+
+func (r *RuntimeClass) SetExternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	runtimeClassInformer := f.Node().V1beta1().RuntimeClasses()
+	r.SetReadyFunc(runtimeClassInformer.Informer().HasSynced)
+	r.lister = runtimeClassInformer.Lister()
+}
+
+func (r *RuntimeClass) SetExternalKubeClientSet(client kubernetes.Interface) {
+	r.client = client
+}
+
+func (r *RuntimeClass) ValidateInitialization() error {
+	if r.lister == nil {
+		return fmt.Errorf("missing RuntimeClass lister")
+	}
+	if r.client == nil {
+		return fmt.Errorf("missing client")
+	}
+	return nil
+}
+
+// Admit makes an admission decision based on the request attributes
+func (r *RuntimeClass) Admit(attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	return admissionAction(r, attributes, "admit")
+}
+
+// Validate makes sure that pod adhere's to RuntimeClass's definition
+func (r *RuntimeClass) Validate(attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	return admissionAction(r, attributes, "validate")
+}
+
+// NewRuntimeClass creates a new always pull images admission control handler
+func NewRuntimeClass() *RuntimeClass {
+	return &RuntimeClass{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+// admissionAction handles Admit and Validate phases of admission, switching based on the admissionPhase parameter
+func admissionAction(r *RuntimeClass, attributes admission.Attributes, admissionPhase string) (err error) {
+
+	// Ignore all calls to subresources or resources other than pods.
+	if shouldIgnore(attributes) {
+		return nil
+	}
+	pod, ok := attributes.GetObject().(*api.Pod)
+	if !ok {
+		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
+	}
+
+	// Get RuntimeClass. Not finding a given runtimeClass does not return an error. We will want to continue in the
+	// case of Validate phase, even if a RuntimeClass is not defined
+	runtimeClass, err := getRuntimeClass(r, pod, pod.Spec.RuntimeClassName)
+	if err != nil {
+		return err
+	}
+
+	switch admissionPhase {
+	case "admit":
+		err := setOverhead(pod, runtimeClass)
+		if err != nil {
+			return err
+		}
+
+	case "validate":
+		err := validateOverhead(pod, runtimeClass)
+		if err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("invalid value for admissionPhase: %s", admissionPhase)
```

Comment:
nit: `admissionPhase` sounds a bit odd to me. the mutating and validating logic should better be separated i suppose? ping @liggitt for suggestions.

Reply:
Yeah, names are hard.  I wanted to deduplicate some of the boiler plate handling that is needed for both mutation and validation phases of the controller. Happy to hear better suggestions for the name.

Reply:
*optional:* Alternatively, just put the boilerplate in a helper:
```
pod, runtimeClass, err := r.prepareObjects(attributes)
```
(needs a better name though)

Reply:
Updated to add the helper.  I named it prepareObjects, but am going to go for a two hour bike ride this evening and think of a more appropriate name.